### PR TITLE
Add test framework for mint

### PR DESCRIPTION
**Pull request by poornas**
Rewrite mint to create a framework for running SDK tests on Docker. 

mc, minio-py, minio-go, minio-java, minio-js sdk tests have currently been added to the framework.

This is w.r.t issue #26

### CODE REVIEW

File: apps/init/initCheck.go [hunk 1]
```diff
@@ -24,9 +24,6 @@ import (
 )
 
 func main() {
-	// obtain endpoint from env.
-	// it is set using `-e` when starting the container
-	// check README.md for instructions.
 	endpoint := os.Getenv("S3_ADDRESS")
```
Comment (harshavardhana) [inline apps/init/initCheck.go]: Remove S3_ADDRESS

File: run.sh [hunk 2]
```diff
@@ -15,72 +15,74 @@
...
+# Setup environment variables for the run.
+_init() {
+	set -e
+
+	# If S3_ADDRESS is not set the tests are run on play.minio.io by default.
+
+	# S3_ADDRESS is passed on as env variables while starting the docker container.
+	# see README.md for info on options.
+	#  Note: https://play.minio.io hosts publicly available Minio server.
+	if [ -z "$S3_ADDRESS" ]; then
```
Comment (harshavardhana) [inline run.sh]: Remove S3_ADDRESS

File: apps/mc/run.sh [hunk 3]
```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
```
Comment (harshavardhana) [inline apps/mc/run.sh]: Avoid binary files..

File: apps/minio-java/run.sh [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    build >>$logfile  2>&1 || { echo "minio-java build failed."; exit 1;}
+ 
+    # run the tests
+    rc=0
+    run 2>>$errfile 1>>$logfile && cleanUp || { echo "minio-java run failed."; rc=1; }
+    return $rc
+}
+
+# invoke the script
+main "$@"
```
Comment (harshavardhana) [inline apps/minio-java/run.sh]: add a newline.

File: README.md [hunk 5]
```diff
@@ -1,51 +1,82 @@
...
+
 The project will be published in Docker hub after further more testing. Till then the docker image has to be built locally and run.
 
 ### Build
 
 ```sh
 $ git clone https://github.com/minio/mint.git
 $ cd mint
-$ docker build -t minio/mint:alpha .
+$ docker build -t minio/mint:mint .
```
Comment (harshavardhana) [inline README.md]: Keep the image as `latest`  `minio/mint:latest`

File: Dockerfile [hunk 6]
```diff
@@ -1,17 +1,29 @@
...
+  python3-pip && \
+  pip3 install yq && \
+  update-alternatives --config java && update-alternatives --config javac
 
-RUN \
-       apk add --no-cache bash git openssh mailcap && \
-       go get -u github.com/minio/minio-go && \
-       chmod +x run.sh
+
+COPY . /mint
```
Comment (harshavardhana) [inline Dockerfile]: Keep two dockerfiles one more Dockerfile and another for release.  Dockerfile.release

File: Dockerfile [hunk 7]
```diff
@@ -1,17 +1,29 @@
...
+ENV PATH $PATH:$GOPATH/bin
 
-# Install commonly used dependencies here.
-# Any unique dependencies can be installed in respective build.sh files 
+RUN apt-get update && apt-get install -yq \
+  curl \
+  default-jre \ 
+  default-jdk \
+  git \
+  golang-go \
```
Comment (harshavardhana) [inline Dockerfile]: go-1.6 version should be moved to go1.7

File: Dockerfile [hunk 8]
```diff
@@ -1,17 +1,26 @@
...
+  python3-pip && \
+  pip3 install yq && \
+  curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash - && \
+  sudo apt-get install -y nodejs && \ 
+  curl -O https://storage.googleapis.com/golang/go1.7.4.linux-amd64.tar.gz && \
+  tar -xf go1.7.4.linux-amd64.tar.gz && \
+  mv go /usr/local && \
+  rm -rf go && \
+  rm go1.7.4.linux-amd64.tar.gz
+COPY . /mint
```
Comment (harshavardhana) [inline Dockerfile]: Are we going to add a new Dockerfile.release which only does `git clone` version?

Comment (poornas) [inline Dockerfile]: yes, we will send an update shortly

**Pull request merged**